cpp/include/raft/sparse/linalg/norm.cuh

```diff
@@ -66,6 +67,37 @@ void csr_row_normalize_max(const int* ia,  // csr row ind array (sorted by row)
   detail::csr_row_normalize_max(ia, vals, nnz, m, result, stream);
 }
 
+/**
+ * @brief Compute row-wise norm of the input matrix and perform fin_op lambda
+ *
+ * Row-wise norm is useful while computing pairwise distance matrix, for
+ * example.
+ * This is used in many clustering algos like knn, kmeans, dbscan, etc...
+ *
+ * @tparam Type the data type
+ * @tparam Lambda device final lambda
+ * @tparam IdxType Integer type used to for addressing
+ * @param norm the output vector of row-wise norm, size [N]
+ * @param ia the input matrix row index array
+ * @param data the input matrix nnz data
+ * @param N number of rows of data
+ * @param type the type of norm to be applied
+ * @param stream cuda stream where to launch work
+ * @param fin_op the final lambda op
+ */
+template <typename Type, typename IdxType = int, typename Lambda = raft::identity_op>
+void rowNormCsr(Type* norm,
```

Comment:
Please accept `raft::device_resources` on public APIs and pass the `cudaStream_t` down to the `detail` namespace. Arguments should be ordered as follows:
1. `device_resources`
2. input pointers
3. output pointers
4. additional parameters

Reply:
All of the pointer-based (dense) public APIs are deprecated in RAFT in favor of the mdspan-based APIs. We're going to be scraping through the sparse public APIs and using the new sparse types everywhere. While I'd prefer that we no longer create pointer-based sparse APIs, I do understand that we don't yet have an alternative at the moment. Still, I'd prefer to at least keep the ordering of new APIs consistent w/ the agreed-upon ordering for consistency (the dense norm computation is both deprecated and didn't yet implement the ordering.). We should always accept `device_resources` in new APIs instead of `cudaStream_t` directly.